app/src/main/java/com/firebase/uidemo/ChooserActivity.java

```diff
@@ -43,6 +46,14 @@
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        if (AuthUI.canHandleIntent(getIntent())) {
+            Intent intent = new Intent(ChooserActivity.this, AuthUiActivity
+                    .class);
+            intent.putExtra(ExtraConstants.EMAIL_LINK_SIGN_IN, getIntent().getData().toString());
+            startActivity(intent);
```

Comment:
You should call `finish()` and `return` here to stop the `ChooserActivity` from ever being displayed in this case.

Reply:
Done.